src/scene/graph-node.js

```diff
@@ -23,6 +23,50 @@ const matrix = new Mat4();
 const target = new Vec3();
 const up = new Vec3();
 
+/**
+ * Helper function that handles signature overloading to receive a test function.
+ *
+ * @param {FindNodeCallback|string} attr - Attribute or lambda.
+ * @param {*} [value] - Optional value in case of `attr` being a `string`
+ * @returns {FindNodeCallback} Test function that receives a GraphNode and returns a boolean.
+ * @ignore
+ */
+function createTest(attr, value) {
+    if (attr instanceof Function) {
+        return attr;
+    }
+    return (node) => {
+        let x = node[attr];
+        if (x instanceof Function) {
+            x = x();
+        }
+        return x === value;
+    };
+}
+
+/**
+ * Helper function to recurse findOne without calling createTest constantly.
+ *
+ * @param {GraphNode} node - Current node.
+ * @param {FindNodeCallback} test - Test function.
+ * @returns {GraphNode|null} A graph node that match the search criteria. Returns null if no
+ * node is found.
+ * @ignore
+ */
+function findNode(node, test) {
+    if (test(node))
+        return node;
+
+    const len = node._children.length;
+    for (let i = 0; i < len; ++i) {
+        const result = findNode(node._children[i], test);
```

Comment:
Since in this code we're going for _super speed_ it probably makes sense to cache `node._children` in a local variable above.

Reply:
Couldn't say how much sense it makes nowadays as it may depends on the JS engine but it probably doesn't hurt (on my end it's sometimes faster sometimes not, always by a very small margin). I also noticed `forEach` weren't caching length and using `i++` instead of `++i` (which on my end seems to make no difference on chrome but sometimes a big one on firefox).